extensions-core/s3-extensions/src/main/java/io/druid/storage/s3/S3StorageDruidModule.java

```diff
@@ -98,16 +101,9 @@ public void configure(Binder binder)
 
   @Provides
   @LazySingleton
-  public AWSCredentialsProvider getAWSCredentialsProvider(final AWSCredentialsConfig config)
```

Comment:
It will be bound in `AWSModule` but the configs come from the same place it always has. The difference is just in the binding mechanics of being able to override.

Previously `AWSCredentialsProvider` was bound and used in the s3 extension, but all the dependencies were created in `AWSModule` anyways, so all it really did was prevent `AWSCredentialsProvider` from being overridden in any other module.

If the s3 extension needs its own `AWSCredentialProvider` that explicitly forbids other modules for overriding it, it can use annotations to specify such a case, and this PR's implementation would effectively be a `@Global`. But until such a use case comes up, I'm not inclined to make guesses that can add complication.